src/components/visualizations/chart/legend/ColorLegend.tsx

```diff
@@ -0,0 +1,73 @@
+// (C) 2007-2020 GoodData Corporation
+import * as React from "react";
+import * as classNames from "classnames";
+import {
+    IColorLegendBox,
+    IColorLegendConfig,
+    IHeatmapLegendLabel as IColorLegendLabel,
+    getColorLegendConfiguration,
+} from "./helpers";
+import { TOP } from "./PositionTypes";
+import { IColorLegendItem } from "../../typings/legend";
+
+export interface IColorLegendProps {
+    colorData: IColorLegendItem[];
+    isSmall: boolean;
+    format?: string;
+    numericSymbols: string[];
+    position: string;
+}
+
+function ColorLabels(labels: IColorLegendLabel[]): JSX.Element {
+    return (
+        <div className="labels">
+            {labels.map(
+                (item: IColorLegendLabel): JSX.Element => {
+                    const { key, label, style } = item;
+                    return (
+                        <span key={key} style={style}>
+                            {label}
+                        </span>
+                    );
+                },
+            )}
+        </div>
+    );
+}
+
+function ColorBoxes(boxes: IColorLegendBox[]): JSX.Element {
+    return (
+        <div className="boxes">
+            {boxes.map(
+                (box: IColorLegendBox): JSX.Element => {
+                    const classes = classNames("box", box.class);
+                    const { key, style } = box;
+                    return <span className={classes} key={key} style={style} />;
+                },
+            )}
+        </div>
+    );
+}
+
+export function ColorLegend(colorLegendProps: IColorLegendProps): JSX.Element {
+    const { colorData, format, numericSymbols, isSmall, position } = colorLegendProps;
+
+    const config: IColorLegendConfig = getColorLegendConfiguration(
+        colorData,
+        format,
+        numericSymbols,
+        isSmall,
+        position,
+    );
+    const classes = classNames(...config.classes);
+    const renderLabelsFirst = config.position === TOP;
+    const { boxes, labels } = config;
+
+    return (
+        <div className={classes}>
+            {renderLabelsFirst && ColorLabels(labels)}
+            {ColorBoxes(boxes)}
```

Comment:
I see this is not React convention.
It should be 
{renderLabelsFirst && <ColorLabels labels={labels}/>
< ColorBoxes boxes={boxes}/>